app/namespaces/data_provider/data_provider_controller.py

```diff
@@ -78,9 +89,19 @@ def post(self):
         if not columns or ("pin_latitude" in columns and "pin_longitude" in columns):
             result = jitter_pins(result)
 
-        # Only paginate if all the pagination parameters have been specified
-        if page_index is not None and per_page is not None and sorting_key is not None and reverse is not None:
-            result = get_paginated_records(result, sorting_key, page_index, per_page, reverse)
+        kwargs = {
+            "records": result,
+            "min_page_index": min_page_index,
+            "max_page_index": max_page_index,
+            "per_page": per_page,
+            "reverse": reverse,
+            "sorting_key": sorting_key
+        }
+
+        kwargs_not_none = { k: v for k, v in kwargs.items() if v is not None }
```

Comment:
super nit but can you change the spacing so that it's {k:v for k, v in kwargs.items() if v is not None }